ConfigSpace/conditions.py

```diff
@@ -47,6 +49,17 @@ def __init__(self) -> None:
     @abstractmethod
     def __repr__(self) -> str:
         pass
+    @abstractmethod
+    def set_vector_idx(self, hyperparameter_to_idx) -> None:
```

Comment:
Could you please check that all the arguments to the functions have a type annotation as well?